core/src/util/builders.cpp

```diff
@@ -59,22 +45,72 @@ void Builders::buildPolygon(const Polygon& _polygon, float _height, PolygonBuild
         }
     }
 
-    // call the tesselator
-    glm::vec3 normal(0.0, 0.0, 1.0);
+    _ctx.earcut(_polygon);
+
+    size_t sumPoints = 0;
+    for (auto& line : _polygon) {
+        sumPoints += line.size();
+    }
 
-    _ctx.numVertices += _ctx.earcut.vertices.size();
+    size_t sumVertices = 0;
+    _ctx.used.assign(sumPoints, 0);
+    for (auto i : _ctx.earcut.indices) {
```

Comment:
I've ported the dropDuplicateVertices method that was previously done in earcut. It's at least the start or endpoint that are duplicate in our polygons (one of them is unused depending on the winding)